src/shared/agent_op.c

```diff
@@ -367,7 +367,8 @@ int set_agent_multigroup(char * group) {
     OS_SHA256_String(group,multi_group_hash);
     char _hash[9] = {0};
 
-    strncpy(_hash,multi_group_hash,8);
+    strncpy(_hash,multi_group_hash,9);
+    _hash[8] = '\0';
     snprintf(multigroup_path, PATH_MAX, "%s/%s" , MULTIGROUPS_DIR, _hash);
```

Comment:
This operation is not even necessary.
This limited copy operation could be done directly in snprintf
```suggestion
    snprintf(multigroup_path, PATH_MAX, "%s/.8%s" , MULTIGROUPS_DIR, multi_group_hash);
```

Reply:
Hi @jnasselle 

You are right, I just correct the syntax to limit the string in `sprintf`. 
`snprintf(multigroup_path, PATH_MAX, "%s/%.8s" , MULTIGROUPS_DIR, multi_group_hash);`

Best regards
Hanes